test/parallel/test-runner-coverage.js

```diff
@@ -37,64 +37,120 @@ function getCoverageFixtureReport() {
+    '13, 16, 17, 18, 19, 20, 21, 22, 27, 39, 43, 44, 61, 62, 66, 67, 71, 72',
+    '\u2139 test/fixtures/test-runner/invalid-tap.js | 100.00 | 100.00 | 100.00 | ',
+    '\u2139 test/fixtures/v8-coverage/throw.js | 71.43 | 50.00 | 100.00 | 5, 6',
+    '\u2139 all files | 78.35 | 43.75 | 60.00 |',
+    '\u2139 end of coverage report',
+  ].join('\n');
 
-test('handles the inspector not being available', (t) => {
-  if (process.features.inspector) {
-    return;
+  if (common.isWindows) {
+    return report.replaceAll('/', '\\');
   }
 
-  const fixture = fixtures.path('test-runner', 'coverage.js');
-  const args = ['--experimental-test-coverage', fixture];
-  const result = spawnSync(process.execPath, args);
+  return report;
+}
 
-  assert(!result.stdout.toString().includes('# start of coverage report'));
-  assert(result.stderr.toString().includes('coverage could not be collected'));
-  assert.strictEqual(result.status, 0);
-  assert(!findCoverageFileForPid(result.pid));
-});
+test('test coverage report', async (t) => {
+  await t.test('--experimental-test-coverage and --test cannot be combined', () => {
+    // TODO(cjihrig): This test can be removed once multi-process code coverage
+    // is supported.
+    const args = ['--test', '--experimental-test-coverage'];
+    const result = spawnSync(process.execPath, args);
+
+    // 9 is the documented exit code for an invalid CLI argument.
+    assert.strictEqual(result.status, 9);
+    assert.match(
+      result.stderr.toString(),
+      /--experimental-test-coverage cannot be used with --test/
+    );
+  });
 
-test('coverage is reported and dumped to NODE_V8_COVERAGE if present', (t) => {
-  if (!process.features.inspector) {
-    return;
-  }
+  await t.test('handles the inspector not being available', (t) => {
+    if (process.features.inspector) {
+      return;
+    }
 
-  const fixture = fixtures.path('test-runner', 'coverage.js');
-  const args = ['--experimental-test-coverage', fixture];
-  const options = { env: { ...process.env, NODE_V8_COVERAGE: tmpdir.path } };
-  const result = spawnSync(process.execPath, args, options);
-  const report = getCoverageFixtureReport();
+    const fixture = fixtures.path('test-runner', 'coverage.js');
+    const args = ['--experimental-test-coverage', fixture];
+    const result = spawnSync(process.execPath, args);
 
-  assert(result.stdout.toString().includes(report));
-  assert.strictEqual(result.stderr.toString(), '');
-  assert.strictEqual(result.status, 0);
-  assert(findCoverageFileForPid(result.pid));
+    assert(!result.stdout.toString().includes('# start of coverage report'));
+    assert(result.stderr.toString().includes('coverage could not be collected'));
+    assert.strictEqual(result.status, 0);
+    assert(!findCoverageFileForPid(result.pid));
+  });
 });
 
-test('coverage is reported without NODE_V8_COVERAGE present', (t) => {
-  if (!process.features.inspector) {
-    return;
-  }
+test('test tap coverage reporter', async (t) => {
```

Comment:
Not something that needs to be done in this PR, but we should probably specify `--test-reporter tap` explicitly for these tests now that the default is sometimes TAP and sometimes spec.

Reply:
> Not something that needs to be done in this PR, but we should probably specify `--test-reporter tap` explicitly for these tests now that the default is sometimes TAP and sometimes spec.

Since `tap` is more machine friendly and `spec` more user friendly, I think the latest one should be used by default always, and being `tap` and opt-in, similar to `json` output.